internal/test/providers/aws/aws.go

```diff
@@ -0,0 +1,404 @@
+	"github.com/aws/aws-sdk-go/service/iam"
+	mapi "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	"github.com/openshift/windows-machine-config-bootstrapper/internal/test/clusterinfo"
+	"k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	awsprovider "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsprovider/v1beta1"
+)
+
+const (
+	infraIDTagKeyPrefix = "kubernetes.io/cluster/"
+	infraIDTagValue     = "owned"
+	windowsLabel        = "machine.openshift.io/os-id"
+	// instanceType is the AWS specific instance type to create the VM with
+	instanceType = "m5a.large"
+)
+
+type awsProvider struct {
+	// imageID is the AMI image-id to be used for creating Virtual Machine
+	imageID string
+	// instanceType is the flavor of VM to be used
+	instanceType string
+	// A client for IAM.
+	iam *iam.IAM
+	// A client for EC2. to query Windows AMI images
+	ec2 ec2iface.EC2API
+	// openShiftClient is the client of the existing OpenShift cluster.
+	openShiftClient *clusterinfo.OpenShift
+	// region in which the Machine needs to be created
+	region string
+	// sshKeyPair is the key pair associated with the Windows VM
+	sshKeyPair string
+}
+
+// newSession uses AWS credentials to create and returns a session for interacting with EC2.
+func newSession(credentialPath, credentialAccountID, region string) (*awssession.Session, error) {
+	if _, err := os.Stat(credentialPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("failed to find AWS credentials from path '%v'", credentialPath)
+	}
+	return awssession.NewSession(&aws.Config{
+		Credentials: credentials.NewSharedCredentials(credentialPath, credentialAccountID),
+		Region:      aws.String(region),
+	})
+}
+
+// newAWSProvider returns the AWS implementations of the Cloud interface with AWS session in the same region as OpenShift Cluster.
+// credentialPath is the file path the AWS credentials file.
+// credentialAccountID is the account name the user uses to create VM instance.
+// The credentialAccountID should exist in the AWS credentials file pointing at one specific credential.
+func newAWSProvider(openShiftClient *clusterinfo.OpenShift, credentialPath,
+	credentialAccountID, instanceType, region, sshKeyPair string) (*awsProvider, error) {
+	session, err := newSession(credentialPath, credentialAccountID, region)
+	if err != nil {
+		return nil, fmt.Errorf("could not create new AWS session: %v", err)
+	}
+
+	ec2Client := ec2.New(session, aws.NewConfig())
+
+	iamClient := iam.New(session, aws.NewConfig())
+
+	imageID, err := getLatestWindowsAMI(ec2Client)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get latest Windows AMI: %v", err)
+	}
+
+	return &awsProvider{imageID, instanceType,
+		iamClient,
+		ec2Client,
+		openShiftClient,
+		region,
+		sshKeyPair,
+	}, nil
+}
+
+// SetupAWSCloudProvider creates AWS provider using the give OpenShift client
+// This is the first step of the e2e test and fails the test upon error.
```

Comment:
I dont think `This is the first step of the e2e test and fails the test upon error.` should be said here, instead if it is not clear it should be commented at the call site